### ISSUE TITLE
DllNotFoundException: Unable to load DLL 'StandaloneFileBrowser', MacBook Air, M1, Sonoma 14

### ISSUE THREAD

**Issue by TerekhinAndrei** [gkngkc/UnityStandaloneFileBrowser#140]
MacBook Air, M1, Sonoma 14

DllNotFoundException: Unable to load DLL 'StandaloneFileBrowser'. Tried the load the following dynamic libraries: Unable to load dynamic library 'StandaloneFileBrowser' because of 'Failed to open the requested dynamic library (0x06000000) dlerror() = dlopen(StandaloneFileBrowser, 0x0005): tried: '/Users/username/Library/Developer/Xcode/DerivedData/XCode-cxvyqunnacmacaehpfacmluyjyqa/Build/Products/Release/StandaloneFileBrowser' (no such file), '/usr/lib/system/introspection/StandaloneFileBrowser' (no such file, not in dyld cache), 'StandaloneFileBrowser' (no such file), '/System/Volumes/Preboot/Cryptexes/OSStandaloneFileBrowser' (no such file), '/Users/username/Library/Developer/Xcode/DerivedData/XCode-cxvyqunnacmacaehpfacmluyjyqa/Build/Products/Release/StandaloneFileBrowser.app/Contents/MacOS/Frameworks/StandaloneFileBrowser' (no such file), '/usr/lib/StandaloneFileBrowser' (no such file, not in dyld cache), 'StandaloneFileBrowser' (no such file)
  at SFB.StandaloneFileBrowserMac.OpenFilePanel (System.String title, System.String directory, SFB.ExtensionFilter[] extensions, System.Boolean multiselect) [0x0000b] in /Users/username/Developer/StandaloneFileBrowser/UnityStandaloneFileBrowser/Assets/StandaloneFileBrowser/StandaloneFileBrowserMac.cs:44 
  at SFB.StandaloneFileBrowser.OpenFilePanel (System.String title, System.String directory, SFB.ExtensionFilter[] extensions, System.Boolean multiselect) [0x0000a] in /Users/username/Developer/StandaloneFileBrowser/UnityStandaloneFileBrowser/Assets/StandaloneFileBrowser/StandaloneFileBrowser.cs:51 
  at SFB.StandaloneFileBrowser.OpenFilePanel (System.String title, System.String directory, System.String extension, System.Boolean multiselect) [0x00032] in /Users/username/Developer/StandaloneFileBrowser/UnityStandaloneFileBrowser/Assets/StandaloneFileBrowser/StandaloneFileBrowser.cs:39 
  at BasicSample.OnGUI () [0x0008c] in /Users/username/Developer/StandaloneFileBrowser/UnityStandaloneFileBrowser/Assets/StandaloneFileBrowser/Sample/BasicSample.cs:20

**Comment by pantsofpeace** [comment]
I'm getting the same issue except when using a Linux build in Ubuntu 24.04. I tested it days before in an older version of Ubuntu without issue.

**Comment by hungnt54** [comment]
I just fixed my problem with this
https://github.com/gkngkc/UnityStandaloneFileBrowser/issues/138#issuecomment-1961364146

**Comment by IvanMurzak** [comment]
The same problem on MacOS M1/M2.
Looks like the `StandaloneFileBrowser.dll` is compiled only for x64 architecture. 
@gkngkc Could you please compile it for others CPU architecture? Especially Apple Silicon is really needed.
![image](https://github.com/user-attachments/assets/fd3f687b-b3ad-4964-8ed3-44fefd54b6c0)

**Comment by IamBiswajitSahooOfficial** [comment]
> The same problem on MacOS M1/M2. Looks like the `StandaloneFileBrowser.dll` is compiled only for x64 architecture. @gkngkc Could you please compile it for others CPU architecture? Especially Apple Silicon is really needed. ![image](https://private-user-images.githubusercontent.com/9135028/384521256-fd3f687b-b3ad-4964-8ed3-44fefd54b6c0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwOTM0MTAsIm5iZiI6MTczMjA5MzExMCwicGF0aCI6Ii85MTM1MDI4LzM4NDUyMTI1Ni1mZDNmNjg3Yi1iM2FkLTQ5NjQtOGVkMy00NGZlZmQ1NGI2YzAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjBUMDg1ODMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDRhZDI0MGRlZDhhYmE3MDI0ZDFjYmRhMjBlOTVlMjMwNWU5ZmQwOTVlZWY5MmVhYzM5ZGNkOTkyNWU2OTk1NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Z2zelz3buqPF7a3n4YVdwfn75V32F3J0n-sBHglZ9DM)

Yes please extremely needed the compiled plugins for Apple Silicons.. :\
